src/main/scala/org/bitcoins/core/protocol/script/ScriptSignature.scala

```diff
@@ -410,6 +387,72 @@ object P2PKScriptSignature extends Factory[P2PKScriptSignature] {
   }
 }
 
+sealed trait CLTVScriptSignature extends ScriptSignature {
+  def scriptSig : ScriptSignature
+
+  override def signatures : Seq[ECDigitalSignature] = scriptSig.signatures
+
+  override def hex = scriptSig.hex
+}
+
+object CLTVScriptSignature {
+  private case class CLTVScriptSignatureImpl(scriptSig : ScriptSignature) extends CLTVScriptSignature
+
+  /**
+    * Creates a CLTVScriptSignature out the [[ScriptPubKey]] we are satisfying, a sequence of [[ECDigitalSignature]], and a sequence
+    * of [[ECPublicKey]] needed to satisfy the scriptPubKey. If a [[P2SHScriptPubKey]] is provided, a redeemScript must also be provided.
+    * @return
+    */
+  def apply(scriptPubKey: ScriptPubKey, sigs : Seq[ECDigitalSignature], pubKeys : Seq[ECPublicKey], redeemScript : Option[ScriptPubKey]) : CLTVScriptSignature = scriptPubKey match {
+    case p2pkScriptPubKey : P2PKScriptPubKey => CLTVScriptSignatureImpl(P2PKScriptSignature(sigs.head))
+    case p2pkhScriptPubKey : P2PKHScriptPubKey => CLTVScriptSignatureImpl(P2PKHScriptSignature(sigs.head, pubKeys.head))
+    case multiSigScriptPubKey : MultiSignatureScriptPubKey => CLTVScriptSignatureImpl(MultiSignatureScriptSignature(sigs))
+    case cltvScriptPubKey : CLTVScriptPubKey => apply(cltvScriptPubKey.scriptPubKeyAfterCLTV, sigs, pubKeys, redeemScript)
+    case csvScriptPubKey : CSVScriptPubKey => apply(csvScriptPubKey.scriptPubKeyAfterCSV, sigs, pubKeys, redeemScript)
+    case p2shScriptPubKey : P2SHScriptPubKey =>
+      require(redeemScript.isDefined, "If the underlying scriptSig is a P2SHScriptSignature, a redeemScript must be defined.")
+      val cltvScriptSigBeforeRedeemScript = apply(redeemScript.get, sigs, pubKeys, None)
+      CLTVScriptSignatureImpl(P2SHScriptSignature(cltvScriptSigBeforeRedeemScript, redeemScript.get))
+    case EmptyScriptPubKey => CLTVScriptSignatureImpl(EmptyScriptSignature)
+    case nonstandard : NonStandardScriptPubKey => throw new IllegalArgumentException("A NonStandardScriptSignature cannot be" +
```

Comment:
I'm just leaving a comment here so I can think about this more, not sure if this is the best way to handle this... However we can't really create a program to spend something we don't know how to spend..